### PR TITLE
batches: check for empty ID when fetching batch workspace file by ID

### PR DESCRIPTION
**Pull request by BolajiOlajide**
## Test plan

Check for empty ID string when fetching batch workspace files by ID.

### CODE REVIEW

**Review by Piszmog — APPROVED** [review]

File: enterprise/cmd/frontend/internal/batches/resolvers/resolver.go [hunk 1]
```diff
@@ -1899,6 +1899,10 @@ func (r *Resolver) batchSpecWorkspaceFileByID(ctx context.Context, gqlID graphql
 		return nil, err
 	}
 
+	if batchWorkspaceFileRandID == "" {
```
Comment (eseliger) [inline enterprise/cmd/frontend/internal/batches/resolvers/resolver.go:1902]: We usually return a specific error here: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@es/execute-flag-nocache/-/blob/enterprise/cmd/frontend/internal/batches/resolvers/resolver.go?L1691%3A1-1693%3A3

And test for that here: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@es/execute-flag-nocache/-/blob/enterprise/cmd/frontend/internal/batches/resolvers/resolver_test.go?L44

**Review by Piszmog — APPROVED** [review]
Add a test case

File: enterprise/cmd/frontend/internal/batches/resolvers/resolver.go (hunk 1 above)
Comment (BolajiOlajide) [reply enterprise/cmd/frontend/internal/batches/resolvers/resolver.go:1902]: Hmmm. 
That makes sense. However, it's not consistent with all the other node queries. I'll add this error for the others.

Comment (eseliger) [reply enterprise/cmd/frontend/internal/batches/resolvers/resolver.go:1902]: must've been an oversight :/

Comment (BolajiOlajide) [reply enterprise/cmd/frontend/internal/batches/resolvers/resolver.go:1902]: Yeah, I figured. Thanks for pointing this out.

**Pull request merged**